src/node.cc

```diff
@@ -2220,6 +2221,39 @@ void Hrtime(const FunctionCallbackInfo<Value>& args) {
   fields[2] = t % NANOS_PER_SEC;
 }
 
+// Microseconds in a second, used in CPUUsage() below
+#define MICROS_PER_SEC 1000000
+
+// CPUUsage use libuv's uv_getrusage() this-process resource usage accessor,
+// to access ru_utime (user CPU time used) and ru_stime (system CPU time used),
+// which are uv_timeval_t structs (long tv_sec, long tv_usec).
+// Returns those values as Float64 microseconds in the elements of the array
+// passed to the function.
+void CPUUsage(const FunctionCallbackInfo<Value>& args) {
+  uv_rusage_t rusage;
+
+  // Call libuv to get the values we'll return, and set the return code.
+  int err = uv_getrusage(&rusage);
+  args.GetReturnValue().Set(err);
```

Comment:
May consider changing this to:
```cpp
args.GetReturnValue().Set(OneByteString(args.GetIsolate(), uv_strerror(err)));
```
and include that message in your error message. would be useful for debugging.